### DISCUSSION TITLE
Merge drawing layers without losing keyframes?

### DISCUSSION THREAD

**Discussion by Aco72n** [keiwando/toonsquid#513] [Q&A]
Hello, Is there a way to create another drawing layer and then merge that to a drawing layer below without blowing away the key frames that are already on the top layer

https://github.com/keiwando/toonsquid/assets/156031418/a03f09ae-4269-49cf-a977-f74e22eaee07

? Or vice verse? Or just have keyframes apply to all drawing layers within a timeline layer?

Let’s say I create a two keyframes to move a character across the screen but then decide I’d like to add a cup to the characters hand:

Is there a way draw the cup on a separate drawing layer, position the cup behind the hand of the the character (by placing that layer below) so that when I merge the character layer down the cup will remain behind the hand, but travel along with the character according to the “position” key frames that are applied to that character layer?

**Comment by Cat-Butler** [comment]
Moderately quick-ish workaround for simple contexts:
1. Tap on the "Position" label of the Position channel in the timeline to highlight it. 
2. Tap on it again, and select Copy from the pop-up menu.
3. Merge the layers in the Layers panel (yes, the keyframes disappear)
4. Repeat the first two steps but use Paste instead. This will paste the keyframes onto the merged layer.

Alternative (possibly better):
Copy the keyframes for the Position channel (with the black ellipse layer selected, see steps 1 and 2 above). Select the cup layer (or your pink quadrilateral in the example) in the Layers panel. With that non-keyframed drawing layer selected, Paste the keyframes (step 4 above) into the animation/timeline layer for that drawing layer.
This duplicates the keyframes into each drawing layer rather than merging the layers. This might be better, as it gives you more flexibility (e.g. allowing the cup to independently rotate/droop for spilling the contents). It also makes it a lot easier to transfer multiple channel keyframes (position, rotation, scale).

Another possibility:
If you need to do more than one channel, it's possible to drag a multi-frame selection box around multiple channels of the drawing layer keyframes on the timeline, and copy and paste those into the timeline for another selected drawing layer (best done at frame 1 or the earliest appropriate keyframe). That approach is a little bit clumsy in my opinion, or maybe that's just me.

And another:
It's also possible to turn on the "Paint Behind" property of brushes and paint the cup into the keyframed hand layer (duplicate the animation layer first if desired for other use), thus using the existing keyframes without modification.
